File: D - King's Puzzle/sol.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
  long long n;
  cin >> n;
  long long lo = 1, hi = 1e9;
  bool ok = false;
  while (lo <= hi) {
    long long mid = (lo + hi) / 2;
    if (mid * (mid + 1) / 2 == n) {
      ok = true;
      break;
    } else if (mid * (mid + 1) / 2 > n) {
      hi = mid - 1;
    } else {
      lo = mid + 1;
    }
  }
  cout << (ok ? "yes" : "no") << endl;
}

int main(){
  ios::sync_with_stdio(false);
  cin.tie(nullptr);  
  int q;
  q = 1;  
  cin >> q;
  while (q--) {
    solve();
  }
}

File: D - King's Puzzle/sol.py
t = int(input())
for i in range(t):
    n = int(input())
    lo, hi = 1, 1000000000
    ans = False
    while lo <= hi:
        mid = (lo + hi) // 2
        target = mid * (mid + 1) // 2
        if target == n:
            ans = True
            break  
        elif target > n:
            hi = mid - 1
        else:
            lo = mid + 1

    if ans == True:
        print("yes")
    else:
        print("no")
